### PR TITLE
Fix(eos_validate_state): ANTA Refactor BGP test to now also support direct neighbors and IPv6 AFI

### PR DESCRIPTION
**Pull request by carl-baillargeon**
## Change Summary

Refactor `AvdTestBGP` to now also support direct neighbors and IPv6 in addition of peer groups.

## Related Issue(s)

N/A

## Component(s) name

`arista.avd.eos_validate_state` **ANTA**

## Proposed changes
`AvdTestBGP` will now test all activated neighbors and peer groups in the IPv4, IPv6 and EVPN address families of VRF default. This will NOT test BGP neighbors configured in a VRF (coming soon in a separate PR).

## How to test

You can add custom structured_config for BGP under specific nodes:
```yaml
l3leaf:  
  node_groups:
    - group: DC2_BL
      mlag: false
      structured_config:
        router_bgp:
          address_family_ipv4:
            neighbors:
            - ip_address: 123.1.1.10
              activate: true
            - ip_address: 123.1.1.11
              activate: true
          neighbors:
          - ip_address: 123.1.1.10
            remote_as: '1234'
            description: External IPv4 BGP peer
          - ip_address: 123.1.1.11
            remote_as: '1234'
            description: External IPv4 BGP peer
          - ip_address: fd5a:fe45:8831:06c5::a
            remote_as: '12345'
          - ip_address: fd5a:fe45:8831:06c5::b
            remote_as: '12345'
          address_family_ipv6:
            neighbors:
            - ip_address: fd5a:fe45:8831:06c5::a
              activate: true
            - ip_address: fd5a:fe45:8831:06c5::b
              activate: true
```
`eos_validate_state` in ANTA mode will now test the additional activated IPv4, IPv6 neighbors.

Also see `eos_validate_state` molecule scenario.

### User Checklist

- N/A

### Repository Checklist

- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [X] My change requires a change to the documentation and documentation have been updated accordingly.
- [X] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by carlbuchmann — COMMENTED** [review]
LGTM, IPv4 works as expected in my lab. Need to test IPv6!

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py [hunk 1]
```diff
@@ -63,75 +63,88 @@ def add_test(mapping: list, description: str) -> None:
...
+
+    def create_tests(self, afi: str, safi: str | None = None) -> None:
+        """Create BGP tests for the given AFI and SAFI."""
+        bgp_neighbors = get(self.structured_config, "router_bgp.neighbors", [])
+
+        # Retrieve peer groups and direct neighbors.
+        peer_groups = get(self.structured_config, f"router_bgp.address_family_{afi}.peer_groups", [])
+        direct_neighbors = get(self.structured_config, f"router_bgp.address_family_{afi}.neighbors", [])
+
+        # Only explicitely activated neighbors and peer groups are tested.
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py]:
```suggestion
        # Only explicitly activated neighbors and peer groups are tested.
```

**Review by MaheshGSLAB — APPROVED** [review]
LGTM
Only one type

File: ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py (hunk 1 above)
Comment (carl-baillargeon) [reply ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestrouting.py]: Fixed in 8a69d425679aa78680263e5e3429932f4d4a4adf

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**